data-prepper-plugins/grok-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepper.java

```diff
@@ -221,6 +251,12 @@ private void matchAndMerge(final Map<String, Object> recordMap) {
         } else {
             mergeCaptures(recordMap, grokkedCaptures);
         }
+
+        if (grokProcesingMatchSuccess) {
+            grokProcessingMatchSuccessCounter.increment();
+        } else {
+            grokProcessingMatchFailureCounter.increment();
+        }
```

Comment:
You can eliminate the grokProcessingMatchSuccess boolean by just checking if grokkedCaptures is not empty

```
if (grokkedCaptures.isEmpty()) {
    grokProcessingMatchFailureCounter.increment();
} else {
    grokProcessingMatchSuccessCounter.increment();
}
```